view_internal_test.go

```diff
@@ -78,43 +78,39 @@ func TestView_DeleteFragment(t *testing.T) {
 }
 
 // Ensure view closes fragment after failed shard broadcast.
-func TestView_CreateFragmentError(t *testing.T) {
+func TestView_CreateFragmentRace(t *testing.T) {
+	var creates errgroup.Group
 	v := mustOpenView("i", "f", "v")
 	defer v.close()
 
 	// Use a broadcaster which intentionally fails.
-	v.broadcaster = errorBroadcaster{}
+	v.broadcaster = delayBroadcaster{delay: 10 * time.Millisecond}
 
 	shard := uint64(0)
 
-	// Create fragment (with error on broadcast).
-	fragment, err := v.CreateFragmentIfNotExists(shard)
-	if !strings.Contains(err.Error(), "intentional error") {
-		if err != nil {
-			t.Fatal(err)
-		} else if fragment == nil {
-			t.Fatal("expected fragment")
-		} else {
-			t.Fatal("expected intentional error")
-		}
-	}
-
-	// Set the broadcaster back to no-op.
-	v.broadcaster = nopBroadcaster{}
-
-	// Try to create the fragment again.
-	_, err = v.CreateFragmentIfNotExists(shard)
+	creates.Go(func() error {
+		_, err := v.CreateFragmentIfNotExists(shard)
+		return err
+	})
+	creates.Go(func() error {
+		_, err := v.CreateFragmentIfNotExists(shard)
+		return err
+	})
+	err := creates.Wait()
 	if err != nil {
-		t.Fatal(err)
+		t.Fatalf("unexpected error: %v", err)
 	}
 }
 
-// errorBroadcaster is a broadcaster which always returns an error.
-type errorBroadcaster struct {
+// delayBroadcaster is a broadcaster which delays briefly.
```

Comment:
total nitpick... it's not really briefly right? "delayBroadcaster is a nopBroadcaster with a configurable delay."?

Reply:
that is also true